crates/task-impls/src/network.rs

```diff
@@ -248,6 +248,14 @@ impl<
     }
 
     async fn cancel_subtasks(&mut self) {}
+
+    async fn periodic_task(&self, task_id: String, sender: &Sender<Arc<Self::Event>>) {
+        broadcast_event(Arc::new(HotShotEvent::HeartBeat(task_id)), sender).await;
+    }
+
+    fn get_task_name(&self) -> String {
+        "NetworkEventTask".to_string()
```

Comment:
Hard-coding this is a dangerous game since it'll need to be synced with the name of the struct, and would not error on typos or name mismatches.. Could we potentially pre-define the trait method using something like [type_name](https://doc.rust-lang.org/std/any/fn.type_name.html)? Not sure if this is the best way to do it, but if it works it'd reduce the manual bit here a lot.

Reply:
yes i agree hardcoding isnt the best. let me try out your ideas. thanks

Reply:
let me know what you think about it now, i used type_name as suggested